### PR TITLE
Feat: Hydration Treasury Values

### PR DESCRIPTION
**Pull request by jasmeets-6575**
## Summary by CodeRabbit

- **New Features**
	- Integrated a new hydration API for managing hydration assets in the Latest Treasury Overview component.
	- Introduced a custom component, TreasuryAssetDisplay, for displaying treasury asset information, including values in USDT and USDC.

- **Enhancements**
	- Improved data fetching logic to ensure readiness of both AssetHub and hydration APIs.
	- Expanded rendering logic to conditionally display hydration asset values based on availability.

### CODE REVIEW

File: src/components/Home/overviewData/LatestTreasuryOverview.tsx [hunk 1]
```diff
@@ -28,6 +29,8 @@ const LatestTreasuryOverview = ({ currentTokenPrice, available, priceWeeklyChang
 	const trailColor = theme === 'dark' ? '#1E262D' : '#E5E5E5';
 	const [assethubApi, setAssethubApi] = useState<ApiPromise | null>(null);
 	const [assethubApiReady, setAssethubApiReady] = useState<boolean>(false);
+	const [hydrationApi, setHydrationApi] = useState<ApiPromise | null>(null);
```
Comment (AleemAlam) [inline src/components/Home/overviewData/LatestTreasuryOverview.tsx:32]: Why are we using so many states here create a custom hook to get the data

File: src/components/Home/overviewData/LatestTreasuryOverview.tsx [hunk 2]
```diff
@@ -172,8 +250,10 @@ const LatestTreasuryOverview = ({ currentTokenPrice, available, priceWeeklyChang
 	useEffect(() => {
```
Comment (AleemAlam) [inline src/components/Home/overviewData/LatestTreasuryOverview.tsx:250]: multilple useEffects re-rendered this component  unnecessary

File: src/components/Home/overviewData/LatestTreasuryOverview.tsx [hunk 3]
```diff
@@ -253,7 +333,7 @@ const LatestTreasuryOverview = ({ currentTokenPrice, available, priceWeeklyChang
 												<div className='ml-1 flex items-baseline gap-1 whitespace-nowrap text-xs font-medium'>
 													<span className='text-blue-light-medium dark:text-blue-dark-medium'>Polkadot</span>
 													<span className='ml-1 text-xs text-bodyBlue dark:text-blue-dark-high'>{available.value}</span>
-													<span className='text-[11px] text-blue-light-medium dark:text-blue-dark-medium'>{chainProperties[network]?.tokenSymbol}</span>
+													<span className='text-[12px] text-blue-light-high dark:text-blue-dark-high'>{chainProperties[network]?.tokenSymbol}</span>
```
Comment (AleemAlam) [inline src/components/Home/overviewData/LatestTreasuryOverview.tsx:336]: dont use arbitrary values use tailwind text-base etc...

File: src/components/Home/overviewData/LatestTreasuryOverview.tsx [hunk 4]
```diff
@@ -283,18 +375,51 @@ const LatestTreasuryOverview = ({ currentTokenPrice, available, priceWeeklyChang
 														</div>
 													</>
 												)}
-												{chainProperties?.[network]?.supportedAssets?.[2] && (
+											</div>
+										</div>
```
Comment (AleemAlam) [inline src/components/Home/overviewData/LatestTreasuryOverview.tsx:379]: Split this component into smaller components

**Review by AleemAlam — CHANGES_REQUESTED** [review]

File: src/components/Home/overviewData/LatestTreasuryOverview.tsx [hunk 5]
```diff
@@ -6,103 +6,51 @@ import React, { useEffect, useState } from 'react';
...
-	});
+
 	const [graphData, setGraphData] = useState<IMonthlyTreasuryTally[]>([]);
 
 	const assetValue = formatBnBalance(assethubValues.dotValue, { numberAfterComma: 0, withThousandDelimitor: false, withUnit: false }, network);
 	const assetValueUSDC = formatUSDWithUnits(String(Number(assethubValues.usdcValue) / 1000000));
 	const assetValueUSDT = formatUSDWithUnits(String(Number(assethubValues.usdtValue) / 1000000));
 
+	const hydrationValue = formatBnBalance(hydrationValues.dotValue, { numberAfterComma: 0, withThousandDelimitor: false, withUnit: false }, network);
+	const hydrationValueUSDC = formatUSDWithUnits(String(Number(hydrationValues.usdcValue) / 1000000));
```
Comment (alphainfinitus) [inline src/components/Home/overviewData/LatestTreasuryOverview.tsx:41]: please use BN to do these calculations

File: src/components/Home/overviewData/LatestTreasuryOverview.tsx [hunk 6]
```diff
@@ -6,103 +6,51 @@ import React, { useEffect, useState } from 'react';
...
+	const hydrationValueUSDT = formatUSDWithUnits(String(Number(hydrationValues.usdtValue) / 1000000));
+
 	const totalTreasuryValueUSD = formatUSDWithUnits(
 		String(
-			(tokenValue + parseFloat(assethubValues.dotValue) / 10000000000) * parseFloat(currentTokenPrice.value) +
+			(tokenValue + parseFloat(assethubValues.dotValue) / 10000000000 + parseFloat(hydrationValues.dotValue) / 10000000000) * parseFloat(currentTokenPrice.value) +
```
Comment (alphainfinitus) [inline src/components/Home/overviewData/LatestTreasuryOverview.tsx:46]: same for this

File: src/components/Home/overviewData/TreasuryAssetDisplay.tsx [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+interface TreasuryAssetDisplayProps {
+	title: string;
+	icon: React.ReactNode;
+	value: string;
+	unit: string;
+	valueUSDT?: string;
+	valueUSDC?: string;
+	isLoading: boolean;
+	supportedAssets: any;
+}
```
[the remark this answers is not in the document]
Comment (alphainfinitus) [reply src/components/Home/overviewData/TreasuryAssetDisplay.tsx:17]: @jasmeets-6575

**Review by alphainfinitus — CHANGES_REQUESTED** [review]

**Pull request merged**